HotTips/Engine/VSTipHistoryManager.cs

```diff
@@ -121,13 +115,36 @@ private void StoreExcludedGroupsToSettings()
 
         public void MarkTipGroupAsIncluded(string tipGroupId)
         {
-            if (_excludedTipGroups == null)
-            {
-                InitialiseTipHistoryManager();
-            }
-
             _excludedTipGroups.Remove(tipGroupId);
             StoreExcludedGroupsToSettings();
         }
+
+        public async Task SetCadenceAsync(DisplayCadence cadence)
+        {
+            await SettingsManager.SetValueAsync(TIP_CADENCE, cadence.Name, isMachineLocal: true);
+        }
+
+        public DisplayCadence GetCadence()
+        {
+            var cadenceString = SettingsManager.GetValueOrDefault(TIP_CADENCE, string.Empty);
+            return string.IsNullOrEmpty(cadenceString) ? DisplayCadence.VSStartup : DisplayCadence.FromName(cadenceString);
+        }
+
+        public async Task SetLastDisplayTimeNowAsync() => await SetLastDisplayTimeAsync(DateTime.UtcNow);
+
+        public async Task SetLastDisplayTimeAsync(DateTime dateTime)
+        {
+            await SettingsManager.SetValueAsync(TIP_NEXT_DISPLAY, dateTime, isMachineLocal: true);
+        }
+
+        public DateTime GetLastDisplayTime()
```

Comment:
Is this the LastDisplayTime or the NextDisplayTime?
The value being returned is called "TIP_NEXT_DISPLAY".